### PR TITLE
Fix prop type errors

### LINKED ISSUE #1688: propType warnings in the console the article page
**Describe the bug**
Missing required props to components aren't being passed. 

**To Reproduce**

1. `npm run dev`
2. visit http://localhost:7080/news/articles/cn7769kpk9mo
3. view console

**Expected behavior**
There shouldn't be any propType errors.

**Screenshots**
image

**Additional context**

### LINKED ISSUE #1690: Use psammead-caption@1.1.3
**Is your feature request related to a problem? Please describe.**
Changes to `psammead-caption@1.1.3` were made to reduce excess padding below the caption.

Now that the Caption spacing PR is merged into Psammead https://github.com/bbc/psammead/issues/482, the fix should be pulled into the Simorgh application.

Changes are in version 1.1.3, but Simorgh is only using version 1.0.1 currently

**Describe the solution you'd like**
Bump version of `@bbc/psammead-caption` to `1.1.3`.

**Testing notes**
[Tester to complete]

Dev insight: No tests need to be updated - just a visual regression to check spacing matches designs on Zeplin: https://app.zeplin.io/project/5baa564c05c2eba0d2b89568/screen/5c596fa88971d341fae3fd53

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #1688 
Resolves #1690 

**Overall change:** _Fix all prop types warnings in the console_

**Code changes:**

- Removed isRequired from alternateName propTypes
- Bumped the @bbc/psammead-caption version to get the correct prop definitions, also pulls in the new styling so there are visual changes as per https://github.com/bbc/psammead/issues/482 and bbc/psammead#524
- Defines propTypes for the TimestampBlock
- Remove fixture data of `superscript` as it will not be implemented as a fragment going forward ([details](https://github.com/bbc/simorgh/issues/402#issuecomment-481613056))
- Auto linting on the JSON fixtures I've edited

**Test instructions**
- `npm run dev` there should be no errors in the console relating to propTypes (There are some errors expected for the manifest because it 500's.)
- Caption padding bottom is expected to be reduced
- Any other visual regression should be raised as unexpected (EG: spacing and typography)

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [x] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
     "@bbc/gel-foundations": "1.2.0",
     "@bbc/psammead-assets": "^0.1.9",
     "@bbc/psammead-brand": "^1.0.1",
-    "@bbc/psammead-caption": "^1.0.1",
+    "@bbc/psammead-caption": "^1.1.3",
```
Comment (sareh) [inline package.json]: This PR also fixes this issue https://github.com/bbc/simorgh/issues/1690 since it pulls in `psammead-caption@1.1.3` and associated visual styling changes. 

Could you please add that to the PR and clarify that the styling changes will also be visible in your PR description test notes.

Otherwise, the `psammead-caption` changes can be pulled into a separate PR.

**Comment by pjlee11** [comment]
Since https://github.com/bbc/simorgh/pull/1627 we now render `<p>` tags inside of the `<figcaption>` meaning that we are actually using `<figcaption>` as a wrapper to other HTML rather than rendering text contents inside of it. This was the correct solution because you can’t render multiple `<figcaption>`’s in a `<figure>`. The outcome of this is that the typography CSS is being ignored because both `Caption` defines typography and then`Paragraph` defines different typography styles.

I think there are a few solutions to this and wanted to know everyones opinion:
1) We make Paragraph typography agnostic and pass in the typography CSS as a prop (similar to how `psammead-timestamp` is currently working)
2) We utilise the styled-component’s `as` functionality and in simorgh have:
```
const CaptionParagraph = styled.p`
  // styles go here 
`;

  <Paragraph
    script={script}
    as={CaptionParagraph}
  />
```
3) Make `script` optional in `psammead-paragraph` so that the Caption’s CSS is used. by inheritance


PS: The current caption styling is only correct because we aren’t passing the required `script` which then causes a propType warning on `npm run dev`

**Comment by pjlee11** [comment]
Following on from the above: 

Option 1 will work for all use cases but means we have to explicitly pass in a “typography” prop anywhere we render Paragraph. This then means for open source use of this component the application using it needs to pass a callback function to define the typography for the component. This feels like something that is very much a bespoke requirement for our application rather than for the component itself.

Option 2 means we are making a bespoke component in Simorgh, (or this could be in Psammead but I don’t think it has much use case to be split out), which means any override going forward means a bespoke component in Simorgh.

Option 3 could be a blueprint for creating a component which doesn’t explicitly need styling because it inherits from the parent (EG: <body>, or <figcaption> etc). This could cause bugs if the inheritance is misconfigured. If used right this would mean there is less CSS for the browser to compute but in larger teams bugs are more likely so the risk might outweigh the benefit

File: src/app/containers/Caption/index.jsx [hunk 2]
```diff
@@ -26,8 +26,11 @@ const chooseOffscreenText = (
       return defaultText;
   }
 };
-const renderParagraph = paragraphBlock => (
-  <Paragraph key={deepGet([0, 'model', 'text'], paragraphBlock)}>
+const renderParagraph = (paragraphBlock, script) => (
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx]:
```suggestion
const renderParagraph = paragraphBlock => (
```

File: src/app/containers/Caption/index.jsx [hunk 3]
```diff
@@ -26,8 +26,11 @@ const chooseOffscreenText = (
...
+  <Paragraph
+    script={script}
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx]:
```suggestion
```

File: src/app/containers/Caption/index.jsx [hunk 4]
```diff
@@ -39,7 +42,7 @@ const renderCaption = (paragraphBlocks, offscreenText, script) => (
     ) : null}
     {paragraphBlocks.map(block => {
       const paragraphBlock = deepGet(['model', 'blocks'], block);
-      return renderParagraph(paragraphBlock);
+      return renderParagraph(paragraphBlock, script);
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx]:
```suggestion
      return renderParagraph(paragraphBlock);
```

**Review by sareh — COMMENTED** [review]
Thanks for that summary, Phil.  
I think option 3 would work best, since we have `psammead-caption@1.1.3` to include styles relating to the `<p>` and `<i>` tags nested within the `<figcaption>` element. 
https://github.com/bbc/psammead/blob/d3f184ee554e32793feed3db65024ddb6db5c875/packages/components/psammead-caption/src/index.jsx#L39-L44

I've therefore suggested the changes to not pass in the script to the Paragraph component, so that the Caption typography styles can be retained.

**Comment by pjlee11** [comment]
Thinking this through I think we could have a combination of 1 and 3. Where the component accepts `typography` as a prop but it is optional. This means we have the following flexibility to explicitly set the components typography settings but also allow the component to be rendered with the typography inherited from the parent element. 

So given the `CaptionParagraph` example, we could have the following: 
```
import Caption as `psammead-caption`;
import Paragraph as `psammead-paragraph`;

const Figure = () => (
  <figure> 
    <Caption>
      <Paragraph> // Note that no script or typography value passed means inherit the typography from Caption
        ...
      </Paragraph>
    </Caption>
  </figure>
);
```

And in an example where we want to set the typography and not rely on inheritance:
```
import Caption as `psammead-caption`;
import Paragraph as `psammead-paragraph`;
import { latin } from '@bbc/gel-foundations/scripts';
import { getPica } from '@bbc/gel-foundations/typography';

const Figure = () => (
  <figure> 
    <Caption>
      <Paragraph    
        typographyFunc={getPica}
        script={latin}>
        ...
      </Paragraph>
    </Caption>
  </figure>
);
```

**Comment by sareh** [comment]
@pjlee11 Sounds good. Will you agree that the making of the script field optional can be done separately to this PR? So that we can have the case 1 that you've stated in this PR. 

The second case shouldn't be needed for the caption - otherwise the typography styles for the caption will be overwritten by those for the paragraph. 

(Also, there's also a fourth way to address this issue. We could change caption to using a `<p>` element instead of the paragraph from.  We're currently doing this to handle `<i>` fragments within a caption & ensure that psammead-caption has the relevant styles.  It'd be a simpler, cleaner approach, although there would be some duplication of styles in the page.)

For now, I'm happy to go with the option that you've stated above - a the mix of cases 1 & 3.

**Comment by pjlee11** [comment]
Blocked on https://github.com/bbc/psammead/pull/521

**Comment by pjlee11** [comment]
Given https://github.com/bbc/simorgh/pull/1692#issuecomment-492118340 maybe this highlights that we are trying to mis-use `psammead-paragraph` should we actually be considering having a default `styled.p` as you have suggested? I'm now wondering if caption should try and use `psammead-paragraph` even with the option's mentioned above, would be interested to hear your thoughts

**Comment by sareh** [comment]
@pjlee11 I agree. Looking at the styles added by `psammead-paragraph` it seems like the only values that are not overwritten in `psammead-caption` are these values:
```
padding-bottom: ${GEL_SPACING_TRPL};
margin: 0; /* Reset */
```
It'd make more sense for `psammead-caption` to be updated from this (currently here https://github.com/bbc/psammead/blob/latest/packages/components/psammead-caption/src/index.jsx#L39)
```
  & > p:last-child {
    padding-bottom: 0;
  }
```
to this:
```
  & > p {
    padding-bottom: ${GEL_SPACING_TRPL};
    margin: 0; /* reset */
  }
  & > p:last-child {
    padding-bottom: 0;
  }
```
and then we do not use `psammead-paragraph` at all in the caption.

**Comment by sareh** [comment]
This Psammead Caption PR makes the changes discussed in my previous comment: https://github.com/bbc/psammead/pull/524

**Comment by pjlee11** [comment]
Now blocked on the caption styles being self contained https://github.com/bbc/psammead/pull/524

File: package.json [hunk 5]
```diff
@@ -79,7 +79,7 @@
...
```
Comment (pjlee11) [inline package.json]: Following merge of https://github.com/bbc/psammead/pull/524 this needs to be bumped to `^1.1.5` and snapshots need updating

**Review by sareh — DISMISSED** [review]
👍 Looks great. No more errors in the console. ✨

**Review by Bopchy — DISMISSED** [review]

**Comment by pjlee11** [comment]
The following fixture data asset still have propTypes errors due:

| Asset ID            | Reason           |
| --------------- |:-------------:|
| c0000000003o | Contains video block which doesn't have an accurate propType definition |
| c0000000004o | Contains video block which doesn't have an accurate propType definition |
| c0000000005o | Contains video block which doesn't have an accurate propType definition  |
| c0000000006o | Contains video block which doesn't have an accurate propType definition |
| c0000000009o | Contains video block which doesn't have an accurate propType definition |
| c0000000012o | Contains video block which doesn't have an accurate propType definition | 
| c0000000013o | Contains video block which doesn't have an accurate propType definition | 
| c0000000015o | Contains video block which doesn't have an accurate propType definition | 
| c0000000016o | Contains video block which doesn't have an accurate propType definition | 
| c0000000017o | Contains video block which doesn't have an accurate propType definition | 
| c0000000018o | Contains video block which doesn't have an accurate propType definition | 
| c0000000019o | Contains video block which doesn't have an accurate propType definition | 
| c0000000022o | Contains video block which doesn't have an accurate propType definition | 

The video propType failures will be fixed within https://github.com/bbc/simorgh/issues/1636

**Comment by sareh** [comment]
@pjlee11 Just to clarify - the proptypes for Video are accurate, the data validator/fixture files need to be updated to match: https://github.com/bbc/simorgh/issues/766 

Everything else looks good! Thanks for this.

**Review by sareh — APPROVED** [review]

**Review by Bopchy — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM. Happy for this to be merged.
